src/Symfony/Component/HttpFoundation/StreamedJsonResponse.php

```diff
@@ -0,0 +1,157 @@
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\HttpFoundation;
+
+/**
+ * StreamedJsonResponse represents a streamed HTTP response for JSON.
+ *
+ * A StreamedJsonResponse uses a structure and generics to create an
+ * efficient resource saving JSON response.
+ *
+ * It uses flush after a specified flush size to directly stream the data.
+ *
+ * @see flush()
+ *
+ * @author Alexander Schranz <alexander@sulu.io>
+ *
+ * Example usage:
+ *
+ * function loadArticles(): \Generator
+ *     // some streamed loading
+ *     yield ['title' => 'Article 1'];
+ *     yield ['title' => 'Article 2'];
+ *     yield ['title' => 'Article 3'];
+ * }),
+ *
+ * $response = new StreamedJsonResponse(
+ *     // json structure with generators in which will be streamed
+ *     [
+ *         '_embedded' => [
+ *             'articles' => loadArticles(), // any generator which you want to stream as list of data
+ *         ],
+ *     ],
+ * );
+ */
+class StreamedJsonResponse extends StreamedResponse
+{
+    public const DEFAULT_ENCODING_OPTIONS = JsonResponse::DEFAULT_ENCODING_OPTIONS;
+
+    private int $encodingOptions = self::DEFAULT_ENCODING_OPTIONS;
+
+    /**
+     * @param mixed[]                        $data      JSON Data containing PHP generators which will be streamed as list of data
+     * @param int                            $status    The HTTP status code (200 "OK" by default)
+     * @param array<string, string|string[]> $headers   An array of HTTP headers
+     * @param int                            $flushSize After every which item of a generator the flush function should be called
+     */
+    public function __construct(
+        private readonly array $data,
+        int $status = 200,
+        array $headers = [],
+        private int $flushSize = 100,
+    ) {
+        parent::__construct($this->stream(...), $status, $headers);
+
+        if (!$this->headers->get('Content-Type')) {
+            $this->headers->set('Content-Type', 'application/json');
+        }
+    }
+
+    private function stream(): void
+    {
+        $generators = [];
+        $structure = $this->data;
+
+        array_walk_recursive($structure, function (&$item, $key) use (&$generators) {
+            if ($item instanceof \Generator) {
+                // using uniqid to avoid conflict with eventually other data in the structure
+                $placeholder = uniqid('__placeholder_', true);
+                $generators[$placeholder] = $item;
+
+                $item = $placeholder;
+            }
+        });
+
+        $keys = array_keys($generators);
+
+        $jsonEncodingOptions = \JSON_THROW_ON_ERROR | $this->getEncodingOptions();
+        $structureText = json_encode($structure, $jsonEncodingOptions);
+
+        foreach ($keys as $key) {
+            [$start, $end] = explode('"'.$key.'"', $structureText, 2);
```

Comment:
this logic currently assumes that the keys appear in the json in the order in which there are in `$keys`. Is this guaranteed (and tested with more complex structures with generators are different depths) ?

Reply:
With the new implementation it is guaranteed the order matches and so nothing is required here and can be nested in whatever way you want it.